lit_llama/utils.py

```diff
@@ -0,0 +1,59 @@
+import torch.utils._device
+
+
+class EmptyInitOnDevice(torch.overrides.TorchFunctionMode):
+    def __init__(self, device=None, dtype=None, quantization_mode=None):
+        """
+        Create tensors with given device and dtype and don't run initialization
+           (but instead use "empty tensors", i.e. uninitialized memory).
```

Comment:
I don't see where the creation of empty tensors happens.

Reply:
PyTorch uses torch.empty to create the buffers for modules and then uses the overriden `toch.nn.init.*` to initialize:

https://github.com/pytorch/pytorch/blob/d47a4bf53fcde103ea6ea52eef66d999f707f61f/torch/nn/modules/linear.py#L96-L98